packages/metadata-common/src/MetadataEditor.tsx

```diff
@@ -100,6 +102,32 @@ export class MetadataEditor extends ReactWidget {
         if (!this.name) {
           this.title.label = `New ${this.schemaDisplayName}`;
         }
+        this.sortedFields = [];
+        const categories: string[] = [];
+        // Find categories of all schema properties
+        for (const schemaProperty in this.schema) {
+          if (!this.schema[schemaProperty].uihints) {
+            this.schema[schemaProperty].uihints = {};
```

Comment:
I believe we do something similar to this in `renderField` might be worth updating that to remove any redundant code.